src/riak_kv_pb_object.erl

```diff
@@ -190,8 +193,8 @@ process(#rpbputreq{bucket=B0, type=T, key=K, vclock=PbVC,
     case Result of
         consistent ->
             process(Req#rpbputreq{if_not_modified=undefined,
-                                  if_none_match=consistent},
-                    State);
+                                  if_none_match=undefined},
```

Comment:
Thought about this some more. We could add a `State#state.consistent == false` clause in that function head to force it to fall down to the next clause after we set that flag. Not sure if this would be that much better, since we're adding even more complexity to the control flow, and we never look at `if_not_modified` again anyway...but if that would address your concerns I could see that as a valid option.